receiver/couchdbreceiver/client.go

```diff
@@ -0,0 +1,125 @@
+package couchdbreceiver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+// CouchDBClient defines the basic HTTP client interface.
+type CouchDBClient interface {
+	BuildReq(path string) (*http.Request, error)
```

Comment:
Does this need to be part of the interface? Seems like it's only used internally so maybe it's an unexported function on the `couchDBClient` struct instead.

Reply:
Removed from interface and made into an unexported function